Source/JavaScriptCore/runtime/PropertyTable.h

```diff
@@ -78,52 +78,34 @@ inline unsigned nextPowerOf2(unsigned v)
     return v;
 }
 
-class PropertyTable final : public JSCell {
-
-    // This is the implementation for 'iterator' and 'const_iterator',
-    // used for iterating over the table in insertion order.
-    template<typename T>
-    class ordered_iterator {
-    public:
-        ordered_iterator<T>& operator++()
-        {
-            m_valuePtr = skipDeletedEntries(m_valuePtr + 1, m_endValuePtr);
-            return *this;
-        }
-
-        bool operator==(const ordered_iterator<T>& other) const
-        {
-            return m_valuePtr == other.m_valuePtr;
-        }
-
-        bool operator!=(const ordered_iterator<T>& other) const
-        {
-            return m_valuePtr != other.m_valuePtr;
-        }
-
-        T& operator*()
-        {
-            return *m_valuePtr;
-        }
-
-        T* operator->()
-        {
-            return m_valuePtr;
-        }
-
-        ordered_iterator(T* valuePtr, T* endValuePtr)
-            : m_valuePtr(valuePtr)
-            , m_endValuePtr(endValuePtr)
-        {
-        }
-
-    private:
-        T* m_valuePtr;
-        T* m_endValuePtr;
-    };
+// compact <-> non-compact PropertyTable
+// We need to maintain two things, one is PropertyOffset and one is unsigned index in index buffer of PropertyTable.
+// But both are typically small. It is possible that we can get optimized table if both are fit in uint8_t, that's
+// compact PropertyTable.
+//
+// PropertyOffset can be offseted with firstOutOfLineOffset since we can get out-of-line property easily, but this
+// offset is small enough (64 currently), so that we can still assume that most of property offsets are < 256.
+//
+// 1. If property offset gets larger than 255, then we get non-compact PropertyTable. It requires at least 191 (255 - 64) properties.
+//    In that case, PropertyTable size should be 256 since it is power-of-two.
+// 2. If index gets larger than 255, then we get non-compact PropertyTable. But we are using 0 and 255 for markers. Thus, if we get 253
+//    used counts, then we need to change the table.
+//
+// So, typical scenario is that, once 128th property is added, then we extend the table via rehashing. At that time, we change the
+// table from compact to non-compact mode. We could explore the way to keep this more larger, like, up to 192 by using 75% load factor
+// for small table (including compact table).
```

Comment:
This last sentence is a bit confusing. I know you're exploring this already, maybe we can just leave the comment out for now

Reply:
OK, dropped.